doc/developer/design/20231027_refresh_mvs.md

```diff
@@ -0,0 +1,279 @@
+
+For showing the previous and next refresh of each MV, I'm planning to create a new table in `mz_internal` called `mz_materialized_view_refreshes (materialized_view_id text, last_completed_refresh timestamptz, next_refresh timestamptz)`. I'm thinking to update it in `Controller::record_frontiers` (where `mz_frontiers` is also updated). The values will be calculated as follows:
+- `next_refresh` will usually be the write frontier of the MV's storage collection, unless:
+  - The write frontier is 0. This happens when the first refresh hasn't completed yet, in which case we need to figure out the first refresh: we'll call `RefreshSchedule::round_up_timestamp` on the MV's `as_of`.
+  - The write frontier is `[]`. This happens when the MV is after its last refresh, and will never be refreshed again. We could represent this simply with a `NULL`.
+- `last_completed_refresh` is roughly `RefreshSchedule::round_down_timestamp(next_refresh - 1)`, plus special handling for the `[]` write frontier case. (`round_down_timestamp` will be a new function, which will be the same as `round_up_timestamp`, but backwards.)
+
+For seeing whether the last refresh is being late, the user can run `EXPLAIN TIMESTAMP` in `STRICT SERIALIZABLE` mode, and look at can respond immediately. If it's false, then the last refresh's completion is overdue. Another way to get the same information would be to check if `mz_materialized_view_refreshes.next_refresh < now()`.
+
+For showing cluster schedules, I'm thinking of doing a similar thing as in `mz_materialized_view_refresh_options`, because eventually the `SCHEDULE =` option will allow for specifying multiple schedules. So I'd create a new table in `mz_internal` called `mz_cluster_schedules`, and it would have a row for each non-trivial (non-`MANUAL`) schedule option of each cluster. This would currently be only `SCHEDULE = ON REFRESH`. Columns would be `(cluster_id text, schedule_type text, refresh_rehydration_time_estimate interval)`. In the `schedule_type`, we would currently always have "ON REFRESH". (Eventually, we'll probably also want a `next_scheduled_turn_on`, but this doesn't seem so urgent. It will get more important when we'll be choosing the warmup time automatically.)
```

Comment:
Generally makes sense! Same comment as above though: `on-refresh` vs `ON REFRESH`.

Reply:
I'll change it.

Reply:
Also, I'm changing `mz_cluster_schedules` to include clusters that have the default schedule (`SCHEDULE = MANUAL`), similarly to above where we changed `mz_materialized_view_refresh_strategies` to include `REFRESH ON COMMIT`.